controllers/configmap_controller.go

```diff
@@ -206,6 +197,22 @@ func (r *ConfigMapReconciler) reconcileNodes(ctx context.Context, windowsInstanc
 	return nil
 }
 
+// configureInstances configures all instances that require configuration
+func (r *ConfigMapReconciler) configureInstances(instances []*instances.InstanceInfo) error {
+	for _, instance := range instances {
+		err := r.ensureInstanceIsUpToDate(instance, map[string]string{BYOHAnnotation: "true",
```

Comment:
If configureInstance is calling ensureInstanceIsUpToDate, wouldn't it end up calling deconfigure if upgrade is required. Correct me if I am wrong, my thinking was we had introduced ensureInstanceIsUpToDate to avoid this issue in the previous implementation.

Reply:
@sebsoto I can lgtm once the function is renamed to what we discussed during standup.

Reply:
Thanks @mansikulkarni96 I have made the change